first_order_model/train.py

```diff
@@ -201,19 +202,126 @@ def train(config, generator, discriminator, kp_detector, checkpoint, log_dir, da
     generator_full = GeneratorFullModel(kp_detector, generator, discriminator, train_params)
     discriminator_full = DiscriminatorFullModel(kp_detector, generator, discriminator, train_params)
     
+    for x in dataloader:
+        if use_lr_video or use_RIFE:
+            lr_frame = F.interpolate(x['driving'], lr_size)
+            if train_params.get('encode_video_for_training', False):
+                target_bitrate = train_params.get('target_bitrate', None)
+                quantizer_level = train_params.get('quantizer_level', -1)
+                if target_bitrate == 'random':
+                    target_bitrate = np.random.randint(15, 75) * 1000
+                nr = x.get('time_base_nr', torch.ones(lr_frame.size(dim=0), dtype=int))
+                dr = x.get('time_base_dr', 30000 * torch.ones(lr_frame.size(dim=0), dtype=int))
+                x['driving_lr'] = get_frame_from_video_codec(lr_frame, nr, \
+                        dr, quantizer_level, target_bitrate) 
+            else:
+                x['driving_lr'] = lr_frame
+
+        for k in x:
+            try:
+                x[k] = x[k].cuda()
+            except:
+                pass
+
+        break
+    get_gen_input(generator_full,x)
     vgg_model = Vgg19()
     original_lpips = lpips.LPIPS(net='vgg')
     vgg_face_model = VggFace16()
+    start = total_macs(generator)
+    prune_rate = train_params.get('shrink_rate', 0.02)
+    reduce_amount = start * prune_rate
+    current = start
+    target = start * train_params.get('target_shrink', 0.25)
 
+    is_first_round = True
     if torch.cuda.is_available():
-        generator_full = DataParallelWithCallback(generator_full, device_ids=device_ids)
-        discriminator_full = DataParallelWithCallback(discriminator_full, device_ids=device_ids)
         original_lpips = original_lpips.cuda()
         vgg_model = vgg_model.cuda()
         vgg_face_model = vgg_face_model.cuda()
-    
+
     loss_fn_vgg = vgg_model.compute_loss
     face_lpips = vgg_face_model.compute_loss
+    prune_percent = train_params.get('prune', 0)
+    if  prune_percent != 0:
+        generator = channel_prune(generator, prune_percent)
+        generator_full.generator = generator
+        optimizer_generator = torch.optim.Adam(generator.parameters(), lr=train_params['lr_generator'], betas=(0.5, 0.999))
+
+    reload_gen = train_params.get('shrunk_gen', None)
```

Comment:
what is the point of this parameter? won't you reload anyway if you're doing netadapt?